packages/client/src/api/oAuth/index.ts

```diff
@@ -0,0 +1,89 @@
+import { BaseApi } from '@api/baseApi'
+import { ClientID, OauthSignInRequest } from './types'
+import { Routes } from '@routes/constants'
+import {
+    generateRandomStateStr,
+    removeState,
+    saveState,
+} from '@utils/authentication/oauth'
+import { closePopup } from '@utils/ui/popup'
+
+const BaseURL = 'oauth'
+const OAuthURL = 'https://oauth.yandex.ru/authorize'
+
+const Port = '3000'
+const TestUrl = `http://localhost:${Port}`
+
+export const ErrCodeParam = 'error'
+export const ErrDescParam = 'error_description'
+export const StateParam = 'state'
+export const ErrorNoState = `параметр ${StateParam} отсутствует`
+export const ErrorState = `параметр ${StateParam} не совпадает с переданным на сервер авторизации`
+
+export class OAuthApi extends BaseApi {
+    static readonly redirectUri = `${TestUrl}${Routes.OAuth}`
+    private clientID = 'dc45f725fcd94bb2a11472239798fc42'
+
+    constructor() {
+        super(BaseURL)
+    }
+
+    cleanup() {
+        closePopup()
+        removeState()
+    }
+
+    async getServiceId(abort?: AbortSignal) {
+        const params = new URLSearchParams([
```

Comment:
Вообще Axios сам преобразовывает данные в query string если передать их в params в виде объекта
```suggestion
        const params = {
            redirect_uri: OAuthApi.redirectUri,
        }
```